internal/resource/config/authenticationpolicies/authentication_policies_resource.go

```diff
@@ -0,0 +1,416 @@
+
+			authenticationApiApplicationRef, respDiags := resourcelink.ToState(ctx, authnSelectionTree.AuthenticationApiApplicationRef)
+			diags.Append(respDiags...)
+
+			rootNode, respDiags := authenticationpolicytreenode.ToState(ctx, authnSelectionTree.RootNode)
+			diags.Append(respDiags...)
+
+			authnSelectionTreeAttrValues := map[string]attr.Value{
+				"id":                                 types.StringPointerValue(authnSelectionTree.Id),
+				"name":                               types.StringPointerValue(authnSelectionTree.Name),
+				"description":                        types.StringPointerValue(authnSelectionTree.Description),
+				"enabled":                            types.BoolPointerValue(authnSelectionTree.Enabled),
+				"handle_failures_locally":            types.BoolPointerValue(authnSelectionTree.HandleFailuresLocally),
+				"authentication_api_application_ref": authenticationApiApplicationRef,
+				"root_node":                          rootNode,
+			}
+
+			authenticationSelectionTreeToState, respDiags := types.ObjectValue(authnSelectionTreesAttrTypes, authnSelectionTreeAttrValues)
+			diags.Append(respDiags...)
+
+			authnSelectionTreesToState = append(authnSelectionTreesToState, authenticationSelectionTreeToState)
+		}
+
+		state.AuthnSelectionTrees, respDiags = types.ListValue(types.ObjectType{AttrTypes: authnSelectionTreesAttrTypes}, authnSelectionTreesToState)
+		diags.Append(respDiags...)
+	} else {
+		state.AuthnSelectionTrees = types.ListNull(types.ObjectType{AttrTypes: authnSelectionTreesAttrTypes})
+	}
+
+	state.TrackedHttpParameters = internaltypes.GetStringSet(r.TrackedHttpParameters)
+
+	return diags
+}
+
+func addOptionalAuthenticationPolicyFields(addRequest *client.AuthenticationPolicy, plan authenticationPoliciesModel) error {
+	addRequest.FailIfNoSelection = plan.FailIfNoSelection.ValueBoolPointer()
+
+	addRequest.AuthnSelectionTrees = []client.AuthenticationPolicyTree{}
+	for _, authnSelectionTree := range plan.AuthnSelectionTrees.Elements() {
+		authnSelectionTreeObj, ok := authnSelectionTree.(types.Object)
+		if !ok {
+			return fmt.Errorf("authn_selection_trees must be a list of objects")
+		}
+		authenticationPolicyTree := client.AuthenticationPolicyTree{}
+		authnSelectionTreeObjElements := authnSelectionTreeObj.Attributes()
+		if id, ok := authnSelectionTreeObjElements["id"]; ok {
+			authenticationPolicyTree.Id = id.(types.String).ValueStringPointer()
+		}
+		if name, ok := authnSelectionTreeObjElements["name"]; ok {
+			authenticationPolicyTree.Name = name.(types.String).ValueStringPointer()
+		}
+		if description, ok := authnSelectionTreeObjElements["description"]; ok {
+			authenticationPolicyTree.Description = description.(types.String).ValueStringPointer()
+		}
+		if authenticationApiApplicationRef, ok := authnSelectionTreeObjElements["authentication_api_application_ref"]; ok {
+			authenticationApiApplicationRefObj, ok := authenticationApiApplicationRef.(types.Object)
+			if !ok {
```

Comment:
It's odd that some of these attributes have checks for their types but others don't